src/main/java/seedu/address/logic/commands/ListGroupCommand.java

```diff
@@ -0,0 +1,56 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+import seedu.address.model.studentgroup.StudentGroup;
+
+/**
+ * Lists all student groups of residents.
+ */
+public class ListGroupCommand extends Command {
+
+    public static final String COMMAND_WORD = "list-group";
+
+    public static final String MESSAGE_SUCCESS = "Listed all student groups";
+
+    private static final Logger logger = Logger.getLogger(ListGroupCommand.class.getName());
+
+
+    @Override
+    public CommandResult execute(Model model) {
+        assert model != null : "Model is null";
+        requireNonNull(model);
+        List<Person> residentList = model.getFilteredPersonList();
+        logger.log(Level.INFO, "The number of residents is " + residentList.size());
```

Comment:
Will this work as intended if the user does:
1. `find n/XX` thereby shortening the list
2. `list-group`

Reply:
Good catch, I have updated the logic.